src/server.c

```diff
@@ -132,8 +149,13 @@ void serverLogRaw(int level, const char *msg) {
         } else {
             role_char = (server.masterhost ? 'S':'M'); /* Slave or Master. */
         }
-        fprintf(fp,"%d:%c %s %c %s\n",
-            (int)getpid(),role_char, buf,c[level],msg);
+        if (isatty(fileno(fp))) {
```

Comment:
let's check if there's any performance implication to calling `isatty`, and consider caching it or at the very least doing it only when `log_to_stdout` is set?

Reply:
`isatty()` perform a system call (on Linux) which has a noticeable performance impact. I'll cache it (but just for stdout) since the current code doesn't assume an immutable log file name and calls `open()` per log, so it doesn't feel right to cache it between `open()`s.

Reply:
In theory someone could do `redis-server --logfile /dev/stdout` or even  `redis-server --logfile /dev/tty`. I can't think of a good reason to do that, though.

Reply:
+1 to keeping the optimization at the expense of not detecting a TTY logfile name.